client/consensus/babe/rpc/src/lib.rs

```diff
@@ -0,0 +1,248 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! RPC api for babe.
+
+use sc_consensus_babe::{Epoch, authorship, Config};
+use futures::{FutureExt as _, TryFutureExt as _};
+use jsonrpc_core::{
+	Error as RpcError,
+	futures::future as rpc_future,
+};
+use jsonrpc_derive::rpc;
+use sc_consensus_epochs::{descendent_query, Epoch as EpochT, SharedEpochChanges};
+use sp_consensus_babe::{
+	AuthorityId,
+	BabeApi as BabeRuntimeApi,
+	digests::PreDigest,
+};
+use serde::{Deserialize, Serialize};
+use sc_keystore::KeyStorePtr;
+use sp_api::{ProvideRuntimeApi, BlockId};
+use sp_core::crypto::Pair;
+use sp_runtime::traits::{Block as BlockT, Header as _};
+use sp_consensus::{SelectChain, Error as ConsensusError};
+use sp_blockchain::{HeaderBackend, HeaderMetadata, Error as BlockChainError};
+use std::{collections::HashMap, fmt, sync::Arc};
+
+type FutureResult<T> = Box<dyn rpc_future::Future<Item = T, Error = RpcError> + Send>;
+
+/// Provides rpc methods for interacting with Babe.
+#[rpc]
+pub trait BabeApi {
+	/// Returns data about which slots (primary or secondary) can be claimed in the current epoch
+	/// with the keys in the keystore.
+	#[rpc(name = "babe_epochAuthorship")]
+	fn epoch_authorship(&self) -> FutureResult<HashMap<AuthorityId, EpochAuthorship>>;
+}
+
+/// Implements the BabeRPC trait for interacting with Babe.
+///
+/// Uses a background thread to calculate epoch_authorship data.
+pub struct BabeRPCHandler<B: BlockT, C, SC> {
+	/// shared refernce to the client.
```

Comment:
```suggestion
	/// shared reference to the client.
```

Reply:
Some of these doc strings could be improved with the same suggestions I made above.